src/NavigationHelper/src/NavigationHelper.cpp

```diff
@@ -0,0 +1,187 @@
+        return false;
+    }
+    if (m_navigationReplanningDelay<0)
+    {
+        yError() << "[configure] navigationTriggerLoopRate must be positive, instead is: " << m_navigationReplanningDelay;
+        return false;
+    }
+
+    std::string replanningTriggerPortPortName = m_portPrefix + "/replanning_trigger:o";
+    if (!m_replanningTriggerPort.open(replanningTriggerPortPortName))
+    {
+        yError() << "[NavigationHelper::init] Could not open" << replanningTriggerPortPortName << " port.";
+        return false;
+    }
+
+    m_runThreads = true;
+    m_navigationTriggerThread = std::thread(&NavigationHelper::computeNavigationTrigger, this);
+    return true;
+}
+
+// This thread synchronizes the walking-controller with the navigation stack.
+// Writes on a port a boolean value when to replan the path
+void NavigationHelper::computeNavigationTrigger()
+{
+    if (!m_publishInfo)
+    {   //exit the function if the infos are not needed
+        return;
+    }
+    bool trigger = false;   //flag used to fire the wait for sending the navigation replanning trigger
+    yInfo() << "[NavigationHelper::computeNavigationTrigger] Starting Thread";
+    yarp::os::NetworkClock myClock;
+    if (m_simulationMode)
+    {
+        myClock.open("/clock", "/navigationTriggerClock");
+    }
+    bool enteredDoubleSupport = false, exitDoubleSupport = true;
+    while (m_runThreads)
+    {
+        //double support check
+        {
+            const std::lock_guard<std::mutex> lock(m_updateFeetMutex);
+            if (m_leftInContact.size()>0 && m_rightInContact.size()>0)  //external consistency check
+            {
+                if (m_leftInContact.at(0) && m_rightInContact.at(0))
+                {
+                    if (exitDoubleSupport)
+                    {
+                        enteredDoubleSupport = true;
+                        exitDoubleSupport = false;
+                    }
+                }
+                else
+                {
+                    if (! exitDoubleSupport)
+                    {
+                        trigger = true; //in this way we have only one trigger each exit of double support
+                    }
+                    exitDoubleSupport = true;
+                }
+            }
+            else
+            {
+                trigger = false;
+            }
+            }
+        
+        //send the replanning trigger after a certain amount of seconds
+        if (trigger)
+        {
+            trigger = false;
+            //waiting -> could make it dependant by the current swing step duration
+            //if in simulation we need to sync the clock, we can't use the system clock
+            if (m_simulationMode)
+            {
+                myClock.delay(m_navigationReplanningDelay);
+            }
+            else
+            {
+                std::this_thread::sleep_for(std::chrono::duration<double, std::milli>(m_navigationReplanningDelay*1000));
```

Comment:
Because it was giving me an error, since `std::chrono::milliseconds` its constructor doesn't accept `double`s, but only `int`.
In the latter example an implicit cast is made.

Reply:
I see, looks good then